zetaclient/tss_signer.go

```diff
@@ -112,7 +112,7 @@ func (tss *TSS) Sign(digest []byte, height uint64, chain *common.Chain, optional
 		log.Warn().Msgf("keysign status FAIL posting blame to core, blaming node(s): %#v", ksRes.Blame.BlameNodes)
 
 		digest := hex.EncodeToString(digest)
-		index := fmt.Sprintf("%s-%d", digest, height)
+		index := fmt.Sprintf("%d-%d-%s-%d", chain.ChainId, nonce, digest, height)
```

Comment:
We generally have the code to calculate prefixes in `keys.go`, `x/observer/types/keys.go`
Maybe we could implement:
```
GetBlameIndex(chainID, nonce, digest, height)
GetBlamePrefix(chainID, nonce)
```
to use it here and in the query, so the low level aspect of calculating the prefix index is abstracted